### PR TITLE
Subscription Validation

### PR DESCRIPTION
**Pull request by austinpaine**
added a small function to check if the services being subscribed to are all valid services.  Addressed a warning seen in the server logs when a user logs off that's caused from the forced subscription to the carousel and WW data-sources.  Confluence page added to address and will full fix the root of the warning when we overhaul how widget data-subscriptions work.

### CODE REVIEW

File: src/clj/dashboard_clj/components/websocket.clj [hunk 1]
```diff
@@ -4,7 +4,6 @@
     [clojure.tools.logging :as log]
     [clojure.core.async :as async]
     [vanilla.subscription-manager :as subman]
-     [dashboard-clj.components.system :as system]
```
Comment (austinpaine) [inline src/clj/dashboard_clj/components/websocket.clj:7]: removed unused import

Comment (cawasser) [reply src/clj/dashboard_clj/components/websocket.clj:7]: did you find any other unused 'requires' anywhere else?

File: src/clj/vanilla/subscription_manager.clj [hunk 2]
```diff
@@ -27,12 +28,17 @@
...
+  [sources]
+  (into [] (remove #(nil? (get-servicedep %)) sources)))
+
 
 
 (defn push-data!
   "Gets the websocket send-fn from the system-map to push out the updated data-source"
   [username source]
-  (prn "push-data! " username source)
+  (log/info "push-data! " username source)
```
Comment (cawasser) [inline src/clj/vanilla/subscription_manager.clj:41]: do we really want to log every time data is pushed to a user?

**Review by cawasser — APPROVED** [review]
a few comments for consideration, but the changes look fine

File: src/clj/dashboard_clj/components/websocket.clj (hunk 1 above)
Comment (austinpaine) [reply src/clj/dashboard_clj/components/websocket.clj:7]: none that jump out from the files I've been working in.  Ive been removing them if i notice.

File: src/clj/vanilla/subscription_manager.clj (hunk 2 above)
Comment (austinpaine) [reply src/clj/vanilla/subscription_manager.clj:41]: I was thinking of keeping this in at least for a bit so we have some logging thats viewable from the AWS console and gives a view into whats happening when its running there.

**Review by chadarm — APPROVED** [review]
Seems good to go.

**Review by gpierce3 — APPROVED** [review]
Built/ran: dev/jar/container.  Looks good, no obvious errors.
However, I did notice:
There was no data for the second satellite power in the carousel.

Noticed on server logging that there was a problem with the second power source.
Error was:
ERROR vanilla.db.excel-data - Exception:  Cannot store nil as a value at [:db/add 533 :channel-2-power nil]

There were some log statements about 'assigning a nil user' to a resource.
Error was:
Tried to subscribe a nil user to  [:channel-power-1000-service :channel-power-2000-service]

Problem with Excel file???

Im approving since I don't know if this was due to mods for this PR.

**Pull request merged**